camp_fin/views.py

```diff
@@ -1191,9 +1191,140 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class CandidateDetail(CommitteeDetailBaseView):
+import sys
+
+from django.apps import apps
+from django.http import HttpResponse
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib import messages
+from django.core import serializers
+
+from django_select2 import forms as s2forms
+
+
+def get_generic_fields():
+    """Return a list of all GenericForeignKeys in all models."""
+    generic_fields = []
+    for model in apps.get_models():
+        for field_name, field in model.__dict__.items():
+            if isinstance(field, GenericForeignKey):
+                generic_fields.append(field)
+    return generic_fields
+
+
+def merge_candidates(primary_object, alias_objects):
+    """
+    Merge several model instances into one, the `primary_object`.
+    Use this function to merge model objects and migrate all of the related
+    fields from the alias objects the primary object.
+    """
+    generic_fields = get_generic_fields()
+
+    # get related fields
+    related_fields = list(
+        filter(lambda x: x.is_relation is True, primary_object._meta.get_fields())
+    )
+
+    many_to_many_fields = list(filter(lambda x: x.many_to_many is True, related_fields))
+
+    related_fields = list(filter(lambda x: x.many_to_many is False, related_fields))
+
+    # Loop through all alias objects and migrate their references to the
+    # primary object
+    deleted_objects = []
+    deleted_objects_count = 0
+    for alias_object in alias_objects:
+        # Migrate all foreign key references from alias object to primary
+        # object.
+        for many_to_many_field in many_to_many_fields:
+            alias_varname = many_to_many_field.name
+            related_objects = getattr(alias_object, f"{alias_varname}_set")
```

Comment:
Annoyingly, had to change this single line from the management command in django-extensions for this to work for us, lol

Reply:
is this specific to us, or could this be helpful upstream?

Reply:
I'm not sure! I can open a PR.